FileNotFoundException thrown after enabling ws cache.

### Play Version 
2.6.5

### API 
Scala

### Operating System
Windows 10 & CentOS Linux 7

### JDK 
Oracle 1.8.0_72

### Expected Behavior
WS cache should works normally with the configuration as following:
```
play.ws {
   cache.enabled = true
   cache.cacheManagerURI="file:./conf/ehcache-play-ws-cache.xml"
}
```

### Actual Behavior
It works normally in dev mode on Windows 10, however, FileNotFoundException thrown after deploying to linux server(CentOS 7).
```
Caused by: net.sf.ehcache.CacheException: Error configuring from file:./conf/ehcache-play-ws-cache.xml. Initial cause was ./conf/ehcache-play-ws-cache.xml (No such file or directory)
        at net.sf.ehcache.config.ConfigurationFactory.parseConfiguration(ConfigurationFactory.java:105)
        at org.ehcache.jcache.JCacheCachingProvider.getCacheManager(JCacheCachingProvider.java:97)
        at org.ehcache.jcache.JCacheCachingProvider.getCacheManager(JCacheCachingProvider.java:72)
        at play.api.libs.ws.ahc.OptionalAhcHttpCacheProvider.optionalCache(AhcWSModule.scala:121)
        at play.api.libs.ws.ahc.OptionalAhcHttpCacheProvider.get$lzycompute(AhcWSModule.scala:96)
        at play.api.libs.ws.ahc.OptionalAhcHttpCacheProvider.get(AhcWSModule.scala:95)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get$lzycompute(AhcWSModule.scala:49)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get(AhcWSModule.scala:45)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get(AhcWSModule.scala:39)
...
Caused by: java.io.FileNotFoundException: ./conf/ehcache-play-ws-cache.xml (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:195)
        at java.io.FileInputStream.<init>(FileInputStream.java:138)
        at java.io.FileInputStream.<init>(FileInputStream.java:93)
        at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
        at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
        at java.net.URL.openStream(URL.java:1045)
        at net.sf.ehcache.config.ConfigurationFactory.parseConfiguration(ConfigurationFactory.java:102)
        ... 81 more
```

--- a/documentation/manual/working/commonGuide/configuration/WsCache.md
+++ b/documentation/manual/working/commonGuide/configuration/WsCache.md
@@ -35,10 +35,10 @@ Cache size is limited by the underlying cache implementation.  Play WS will crea
 
 > **NOTE**: If you do not limit the HTTP cache or expire elements in the cache, then you may cause the JVM to run out of memory.
 
-In ehcache, you can specify an existing cache by specifying [CacheManager](https://static.javadoc.io/javax.cache/cache-api/1.0.0/javax/cache/CacheManager.html) URI explicitly, which is used in `cachingProvider.getCacheManager(uri, environment.classLoader)`:
+In ehcache, you can specify an existing cache by specifying [CacheManager](https://static.javadoc.io/javax.cache/cache-api/1.0.0/javax/cache/CacheManager.html) resource explicitly, which is used in `cachingProvider.getCacheManager(uri, environment.classLoader)`:
 
 ```
-play.ws.cache.cacheManagerURI="file:./conf/ehcache-play-ws-cache.xml"
+play.ws.cache.cacheManagerResource="ehcache-play-ws-cache.xml"
 ```
 
 and then adding a cache such as following into the `conf` directory:
@@ -56,6 +56,8 @@ and then adding a cache such as following into the `conf` directory:
 
 </ehcache>
 ```
+
+> **NOTE**: `play.ws.cache.cacheManagerURI` is deprecated, use `play.ws.cache.cacheManagerResource` with a path on the classpath instead.
 
 ## Debugging the Cache
 

--- a/framework/src/play-ahc-ws/src/main/resources/reference.conf
+++ b/framework/src/play-ahc-ws/src/main/resources/reference.conf
@@ -21,9 +21,13 @@ play.ws.cache {
   # A specific caching provider name (e.g. if both ehcache and caffeine are set)
   cachingProviderName = ""
 
-  # The cache manager URI to use, i.e.
-  # cacheManagerURI = "file:./conf/ehcache-play-ws-cache.xml"
-  # Or caffeine:
-  # cacheManagerURI = "file:./conf/application.conf"
-  cacheManagerURI = ""
+  # The CacheManager resource to use. For example:
+  #
+  # cacheManagerResource = "ehcache-play-ws-cache.xml"
+  #
+  # If null, will use the ehcache default URI.
+  cacheManagerResource = null
+
+  # The CacheManager URI to use. If non-null, this is used instead of cacheManagerResource.
+  cacheManagerURI = null
 }

--- a/framework/src/play-ahc-ws/src/main/scala/play/api/libs/ws/ahc/AhcWSModule.scala
+++ b/framework/src/play-ahc-ws/src/main/scala/play/api/libs/ws/ahc/AhcWSModule.scala
@@ -3,21 +3,20 @@
  */
 package play.api.libs.ws.ahc
 
+import java.net.URI
 import javax.cache.configuration.FactoryBuilder.SingletonFactory
 import javax.cache.configuration.MutableConfiguration
 import javax.cache.expiry.EternalExpiryPolicy
 import javax.cache.{ CacheManager, Caching, Cache => JCache }
 import javax.inject.{ Inject, Provider, Singleton }
-
 import akka.stream.Materializer
 import com.typesafe.sslconfig.ssl.SystemConfiguration
 import com.typesafe.sslconfig.ssl.debug.DebugConfiguration
 import play.api.inject.{ ApplicationLifecycle, SimpleModule, bind }
 import play.api.libs.ws._
 import play.api.libs.ws.ahc.cache._
-import play.api.{ Configuration, Environment }
+import play.api.{ Configuration, Environment, Logger }
 import play.shaded.ahc.org.asynchttpclient.{ AsyncHttpClient, DefaultAsyncHttpClient }
-
 import scala.concurrent.{ ExecutionContext, Future }
 
 /**
@@ -113,16 +112,16 @@ class OptionalAhcHttpCacheProvider @Inject() (
           cacheConfig.cachingProviderName match {
             case name if name.nonEmpty =>
               Caching.getCachingProvider(name, environment.classLoader)
-            case other =>
+            case _ =>
               Caching.getCachingProvider(environment.classLoader)
           }
-        cacheConfig.cacheManagerURI match {
-          case uriString: String if uriString.nonEmpty =>
-            cachingProvider.getCacheManager(new java.net.URI(uriString), environment.classLoader)
-          case other =>
+        val cacheManagerURI =
+          cacheConfig.cacheManagerURI match {
+            case uriString if uriString.nonEmpty => new URI(uriString)
             // null means use #getDefaultURI
-            cachingProvider.getCacheManager(null, environment.classLoader)
-        }
+            case _ => null
+          }
+        cachingProvider.getCacheManager(cacheManagerURI, environment.classLoader)
       }
 
       Option(cacheManager).map { cm =>
@@ -184,12 +183,26 @@ class OptionalAhcHttpCacheProvider @Inject() (
     cachingProviderName: String)
 
   object AhcHttpCacheParser {
+    // For the sake of compatibility, parse both cacheManagerResource and cacheManagerURI, use cacheManagerURI first.
+    // Treat cacheManagerResource as a resource on the classpath and convert it to an URI string.
     def fromConfiguration(configuration: Configuration): AhcHttpCacheConfiguration = {
+      val cacheManagerURI = configuration.get[Option[String]]("play.ws.cache.cacheManagerURI") match {
+        case Some(uriString) =>
+          Logger(AhcHttpCacheParser.getClass)
+            .warn("play.ws.cache.cacheManagerURI is deprecated, use play.ws.cache.cacheManagerResource with a path on the classpath instead.")
+          uriString
+        case None =>
+          configuration.get[Option[String]]("play.ws.cache.cacheManagerResource")
+            .filter(_.nonEmpty)
+            .flatMap(environment.resource)
+            .map(_.toURI.toString)
+            .getOrElse("")
+      }
       AhcHttpCacheConfiguration(
         enabled = configuration.get[Boolean]("play.ws.cache.enabled"),
         cacheName = configuration.get[String]("play.ws.cache.name"),
         heuristicsEnabled = configuration.get[Boolean]("play.ws.cache.heuristics.enabled"),
-        cacheManagerURI = configuration.get[String]("play.ws.cache.cacheManagerURI"),
+        cacheManagerURI = cacheManagerURI,
         cachingProviderName = configuration.get[String]("play.ws.cache.cachingProviderName")
       )
     }

--- a/framework/src/play-ahc-ws/src/test/scala/play/api/libs/ws/ahc/OptionalAhcHttpCacheProviderSpec.scala
+++ b/framework/src/play-ahc-ws/src/test/scala/play/api/libs/ws/ahc/OptionalAhcHttpCacheProviderSpec.scala
@@ -31,7 +31,7 @@ class OptionalAhcHttpCacheProviderSpec(implicit ee: ExecutionEnv) extends PlaySp
       val settings = Map(
         "play.ws.cache.enabled" -> "true",
         "play.ws.cache.cachingProviderName" -> classOf[JCacheCachingProvider].getName,
-        "play.ws.cache.cacheManagerURI" -> env.classLoader.getResource("ehcache-play-ws-cache.xml").toURI.toString
+        "play.ws.cache.cacheManagerResource" -> "ehcache-play-ws-cache.xml"
       )
       Configuration.load(env, settings)
     }).build()) {
@@ -46,7 +46,7 @@ class OptionalAhcHttpCacheProviderSpec(implicit ee: ExecutionEnv) extends PlaySp
       val settings = Map(
         "play.ws.cache.enabled" -> "true",
         "play.ws.cache.cachingProviderName" -> classOf[CaffeineCachingProvider].getName,
-        "play.ws.cache.cacheManagerURI" -> "caffeine.conf"
+        "play.ws.cache.cacheManagerResource" -> "caffeine.conf"
       )
       Configuration.load(env, settings)
     }).build()) {

